src/modules/ProductLicense.rb

```diff
@@ -778,6 +785,44 @@ def display_beta(id)
       end
     end
 
+    # Determines whether the license was accepted for a given product
+    #
+    # This method reads the license content and uses the new
+    # Y2Packager::ProductLicense API to determine whether the license was
+    # accepted or not. This is just a transitory method because, in the future,
+    # all licenses should be read through the new API.
+    #
+    # @param product  [Y2Packager::Product] Product instance
+    # @param licenses [Hash<String,String>] License files indexed by language
+    # @return [boolean] true if the license was already accepted; false if it was
+    #   not acceptedd or it is not found.
+    def license_accepted_for?(product, licenses)
+      license_file = licenses["en_US"] || licenses["en"] || licenses[""]
+      content = SCR.Read(path(".target.string"), license_file).to_s
+      if content.empty?
+        log.info "No license was found for #{product.name} in the repository"
+        return false
+      end
+
+      license = Y2Packager::ProductLicense.find(product.name, content: content)
+      license && license.accepted?
+    end
+
+    # Find the product in the given repository
+    #
+    # @param src_id [Integer] Repository ID
+    # @return [Y2Packager::Product,nil] Product or nil if it was not found
+    def repository_product(src_id)
+      product_h = Yast::Pkg.ResolvableProperties("", :product, "").find do |properties|
+        properties["source"] == src_id
+      end
+      if product_h.nil?
+        log.info "No product was found in the repository (#{src_id})" unless product_h
```

Comment:
can this happen at all?

Reply:
and useless `unless`

Reply:
Not for official addons, but I prefer to be on the safe side.